polaris: 0.14.3 -> 0.15.0

pbsds opened the pull request:
Changelog: https://github.com/agersant/polaris/blob/master/CHANGELOG.md


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

pbsds committed:
polaris: 0.14.3 -> 0.15.0

Changelog: https://github.com/agersant/polaris/blob/master/CHANGELOG.md

pbsds committed:
polaris-web: 69 -> 76

Diff: https://github.com/agersant/polaris-web/compare/build-69...build-76

jayay commented:
@pbsds Does the systemd unit work for you? I had to apply this:

```diff
diff --git a/nixos/modules/services/misc/polaris.nix b/nixos/modules/services/misc/polaris.nix
index 834ceb8c1c6a..1e3a0b289ae7 100644
--- a/nixos/modules/services/misc/polaris.nix
+++ b/nixos/modules/services/misc/polaris.nix
@@ -91,10 +91,11 @@ in
       serviceConfig = rec {
         User = cfg.user;
         Group = cfg.group;
-        DynamicUser = true;
+        #DynamicUser = true;
         SupplementaryGroups = cfg.extraGroups;
         StateDirectory = "polaris";
         CacheDirectory = "polaris";
+        WorkingDirectory = "/var/lib/polaris/";
         ExecStart = escapeShellArgs (
           [
             "${cfg.package}/bin/polaris"
@@ -115,8 +116,8 @@ in

         # Security options:

-        #NoNewPrivileges = true; # implied by DynamicUser
-        #RemoveIPC = true; # implied by DynamicUser
+        NoNewPrivileges = true; # implied by DynamicUser
+        RemoveIPC = true; # implied by DynamicUser

         AmbientCapabilities = "";
         CapabilityBoundingSet = "";
@@ -125,7 +126,7 @@ in

         LockPersonality = true;

-        #PrivateTmp = true; # implied by DynamicUser
+        PrivateTmp = true; # implied by DynamicUser
         PrivateDevices = true;
         PrivateUsers = true;

@@ -143,7 +144,7 @@ in
           "AF_UNIX"
         ];
         RestrictRealtime = true;
-        #RestrictSUIDSGID = true; # implied by DynamicUser
+        RestrictSUIDSGID = true; # implied by DynamicUser

         SystemCallArchitectures = "native";
         SystemCallErrorNumber = "EPERM";
```
and also had to put the creation of the user and group into my own config file. I feel like the user and group should be created automatically, if they weren't set to anything other than the default. 🤔

wlcx commented on pkgs/servers/polaris/default.nix:
```diff
@@ -10,13 +10,13 @@
     repo = "polaris";
-    rev = version;
-    hash = "sha256-2GHYIlEzRS7KXahdrxMjyIcPCNw8gXJw5/4ZpB/zT3Y=";
+    tag = version;
```
```suggestion
    tag = finalAttrs.version;
```

wlcx commented on pkgs/servers/polaris/default.nix:
```diff
@@ -10,13 +10,13 @@
 
 rustPlatform.buildRustPackage rec {
```
```suggestion
rustPlatform.buildRustPackage (finalAttrs: {
```

wlcx commented on pkgs/servers/polaris/default.nix:
```diff
@@ -10,13 +10,13 @@
+    tag = version;
+    hash = "sha256-uwYNyco4IY6lF+QSVEOVVhZCJ4nRkj8gsgRA0UydLHU=";
 
     # The polaris version upstream in Cargo.lock is "0.0.0".
```
It seems this isn't the case in the tagged release, so I think all this can go :) https://github.com/agersant/polaris/commit/11989aef00118addb8aa43295e785dd9352dcbd0

pbsds commented:
> Does it start on your machine with DynamicUser = true; in the systemd unit? 

No it did not, and I have not had the time to look into it yet, sorry for the radio silence until now.

pbsds committed:
nixos/test/polaris: expand test

pbsds committed:
nixos/polaris: store writeable config file in StateDirectory

Polaris does not try to write to it during normal startup, but during a migration from old state it tries to write to it, prompting an OS error since the store is read-only.

https://github.com/agersant/polaris/blob/5fe45aa059358677088bfaa4db90d16f3d09f37b/src/app.rs#L271

pbsds committed:
polaris: stop patching version in Cargo.lock

pbsds committed:
polaris-web: drop, move to polaris.web

pbsds committed:
polaris: migrate to finalAttrs

pbsds committed:
nixo/test/polaris: migrate to runTest

pbsds commented:
It still seems to be having some issue, i'll draft in the meanwhile

pbsds commented:
https://github.com/agersant/polaris/issues/245
